pkg/block/s3/adapter.go

```diff
@@ -404,6 +412,11 @@ func (a *Adapter) GetPreSignedURL(ctx context.Context, obj block.ObjectPointer,
 	presigner := s3.NewPresignClient(client,
 		func(options *s3.PresignOptions) {
 			options.Expires = a.preSignedExpiry
+			if a.preSignedEndpoint != "" {
```

Comment:
Please add some mimimal tests for this feature? 
i.e make make sure we get the presign endpoint when configured, and get regular urls when not etc.

Reply:
added